Get Postgres working

dpoirier opened the pull request:
Probably not ready to be merged yet, because it still refers to an argyle branch that's waiting as a pull request against the real repo.

Changes in salt:
- set locale
- add github's public key to ssh_known_hosts

Change in rapidsms-salt fabfile:
- add db user (idempotently)
- add db (idempotently)
  This required some additions to argyle that are pending at https://github.com/mlavin/argyle/pull/13

dpoirier committed:
Get Postgres working

copelco commented on salt/roots/salt/sshd/init.sls:
```diff
@@ -6,12 +6,28 @@ ssh:
+
+
+# Github's server public key
+/etc/ssh/ssh_known_hosts:
```
Might be able to use https://salt.readthedocs.org/en/latest/ref/states/all/salt.states.ssh_known_hosts.html

dpoirier replied:
That almost does it, but seems to only support users' known_hosts files, not the system one. I'd rather just do this once and not have to worry about which users will be ssh'ing to github.

dpoirier replied:
Actually it does work. If you pass in the absolute path where it's wanting the relative path, it'll happily write to that absolute path.  That'll be better than what I had.

dpoirier committed:
Use salt's ssh_known_hosts state

copelco commented on fabfile.py:
```diff
@@ -75,20 +79,53 @@ def know_github():
         roles = SERVER_ROLES
     if 'base' not in roles:
         roles.insert(0, 'base')
+    if 'db' in roles:
```
I was hoping to offload as much of the database setup as possible to salt with the postgres templates, but I haven't thought through the entire problem yet. Can we just get rid of env.db?

dpoirier replied:
I was thinking that setting up a database for our project to use was more a deploy item than provisioning, since we might want to run multiple projects on the same system.  But we could go either way.

copelco replied:
Hmm, good point. Maybe we can actually make both of them work. For example, let Salt setup the initial state: create a UTF8 cluster, create a user, create a database based on what's in pillar. We can use a different state trees to define the database name for each machine, e.g. database_name: rapidsms_staging or something. So the fabfile can just assume the database exists and we set env.db accordingly.

Another thing to think about is whether or not we're going to have the need to manage passwords. Maybe for rabbit? Or if we change the gunicorn user to be www-data. In that case, the fabfile, during setup_server(), would only need to reset the user's password, but otherwise everything else would be managed by salt.

dpoirier committed:
Create database in salt, other cleanup

dpoirier committed:
Don't need to install pip and virtualenv from fabfile because salt already did

dpoirier committed:
Get Vagrant working better

Main change in this commit is to add the `vagrant` user to the `admin` group so that
it can sudo without a password. Vagrant expects that to work and various vagrant
commands fail otherwise.

Also, set ALLOWED_HOSTS in the staging config to wildcard since our nginx config
checks the host already and it was 500'ing.

Also, quiet down some of the operations in the fabfile like `pip install` and
`git clone`.

Also, more comments.

dpoirier committed:
A few more doc updates

dpoirier commented:
@copelco @lemanal Any objection to my going ahead and merging this? It gets things a lot closer to working, not just postgres fixes.

copelco commented:
Looks good :thumbsup:
